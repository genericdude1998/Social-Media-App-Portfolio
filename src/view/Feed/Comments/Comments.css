*{
    box-sizing: border-box;
    margin: 0px;
}

.baseCommentsList{
    list-style: none;
}

.baseComment{
    background-color: gainsboro;
    margin-left: 10px;
    max-width: 100vh;
    word-wrap:break-word;
}

.baseName{
    color: royalblue;
    margin-right: 5px;
}

.baseDate{
   text-align: right;
   margin-right: 20px;
}

.dark{
    background-color: darkred;
    color: white;
}