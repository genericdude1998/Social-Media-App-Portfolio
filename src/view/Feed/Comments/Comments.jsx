import React from 'react';
import { connect } from 'react-redux';
import { mapCommentsDispatchToProps } from '../../mappers/mappers';

const Comments = ({comments, postId}) => {
    return (
        <div>
            <ul>
                {comments.map(comment => {
                    return(
                        <li key={postId}>
                            <h5>
                                {comment.name}
                            </h5>
                            <p>
                                {comment.content}
                            </p>
                            <p>
                                {comment.date}
                            </p>
                        </li>
                    )
                })}   
            </ul>
        </div>
    );
}

export default Comments;
