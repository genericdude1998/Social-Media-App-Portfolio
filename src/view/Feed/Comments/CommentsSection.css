*{
    box-sizing: border-box;
    margin: 0px;
}

.baseCommentsSection{
    background-color:white;
    margin-bottom: 30px;
}

.dark{
    background-color: darkred;
}