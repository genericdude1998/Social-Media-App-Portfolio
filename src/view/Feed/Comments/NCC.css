*{
    box-sizing: border-box;
    margin: 0px;
}

.baseNCC{
    background-color: darkblue;
    height: 50px;
    margin: 0px;
}

.baseNCCForm{
    display: flex;
    height: 100%;
    flex-direction: row;
    justify-content: flex-end;
    margin-right: 15px;
    align-items: center;
}

.baseNCCLabel{
    color: royalblue;
    margin-right: 5px;
}

.baseNCCInput{
    margin-right: 5px;
}

.baseNCCButton{
    border: 1px solid gray;
    padding: 2px;
    border-radius: 1px;
    margin-right: 5px;
}

.dark{
    background-color: darkred;
    color: white;
}