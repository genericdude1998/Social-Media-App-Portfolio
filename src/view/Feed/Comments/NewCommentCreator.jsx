import React from 'react';
import { connect } from 'react-redux';
import { useNavigate } from 'react-router';
import { mapNCCDispatchToProps, mapNCCStateToProps } from '../../mappers/mappers';
import styles from './NCC.css';
import withTheme from '../../theme/withTheme';

const NewCommentCreator = ({onSendNewComment, onSetCommentContent, newCommentError, content, postId, theme}) => {
    const navigate = useNavigate();
    return (
        <div className = {`${styles.baseNCC} ${theme === 'dark' ? styles.dark : null}`}>
            <form className = {styles.baseNCCForm}onSubmit={onSendNewComment(content,navigate, postId)}>
                <label className={`${styles.baseNCCLabel} ${theme === 'dark' ? styles.dark : null}`}>Comment</label>
                <input className={styles.baseNCCInput}type="text" onChange={onSetCommentContent}/>
                <button className={styles.baseNCCButton} type={'submit'}>Send Comment</button>
            </form>
        </div>
    );
}

const ConnectedNCC = connect(mapNCCStateToProps, mapNCCDispatchToProps)(NewCommentCreator)

export default  withTheme(ConnectedNCC);
