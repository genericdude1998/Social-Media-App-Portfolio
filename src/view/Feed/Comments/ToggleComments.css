*{
    box-sizing: border-box;
    margin: 0px;
}

.baseOpenCommentsButton{
    border: 1px solid gray;
    padding: 2px;
    border-radius: 1px;
    margin-left: 5px;
}