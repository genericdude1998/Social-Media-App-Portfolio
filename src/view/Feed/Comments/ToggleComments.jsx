import React from 'react';
import styles from './ToggleComments.css'

const ToggleComments = ({comments, postId, openComments ,onOpenComments, onCloseComments}) => {
    return (
        !openComments ? 
            <button className={styles.baseOpenCommentsButton} onClick={() => onOpenComments(postId)}>Open Comments</button> 
            : 
            <button className={styles.baseOpenCommentsButton} onClick={() => onCloseComments(postId)}>Close Comments</button>
    );
}

export default ToggleComments;
