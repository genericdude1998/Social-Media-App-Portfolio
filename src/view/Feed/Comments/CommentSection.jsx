import React from 'react';
import ToggleComments from './ToggleComments';
import Comments from './Comments';
import ConnectedNCC from './NewCommentCreator';
import { useNavigate } from 'react-router';

const CommentSection = ({comments, postId, openComments, onOpenComments, onCloseComments}) => {
    return (
        <div>
            <ConnectedNCC postId={postId}/>
            {comments.length !== 0 ?
                <ToggleComments comments={comments} postId={postId} openComments={openComments} onOpenComments={onOpenComments} onCloseComments={onCloseComments}/>
                :
                null
            }
            {openComments ? 
                <Comments postId = {postId} comments={comments}></Comments>
                :
                null
            }
        </div>
    );
}

export default CommentSection;
