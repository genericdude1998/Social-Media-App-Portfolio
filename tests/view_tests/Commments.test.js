import React from 'react';
import { shallow } from 'enzyme';
import Comments from '../../src/view/Feed/Comments/Comments';
import {mockComments, mockId} from '../mockValues';

let wrapper;

describe('Comments', () => {
    it('should matchSnapshot when given an array of comments', () => {
        wrapper = shallow(
            <Comments.WrappedComponent
                comments={mockComments}
                postId={mockId}
                theme={'light'}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('should matchSnapshot when given an empty array of comments', () => {
        wrapper = shallow(
            <Comments.WrappedComponent
                comments={[]}
                postId={mockId}
                theme={'light'}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('should matchSnapshot when given a dark theme prop', () => {
        wrapper = shallow(
            <Comments.WrappedComponent 
                comments={mockComments}
                postId={mockId}
                theme={'dark'}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});