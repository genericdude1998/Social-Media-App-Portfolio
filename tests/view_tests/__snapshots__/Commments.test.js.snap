// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments should matchSnapshot when given a dark theme prop 1`] = `
<div>
  <ul
    className="baseCommentsList dark"
  >
    <li
      className="baseComment dark"
      key="0"
    >
      <h5
        className="baseName dark"
      >
        mock_name
      </h5>
      <p
        className="dark"
      >
        mock_content
      </p>
      <p
        className="baseDate dark"
      >
        mock_date
      </p>
    </li>
    <li
      className="baseComment dark"
      key="1"
    >
      <h5
        className="baseName dark"
      >
        mock_name
      </h5>
      <p
        className="dark"
      >
        mock_content
      </p>
      <p
        className="baseDate dark"
      >
        mock_date
      </p>
    </li>
    <li
      className="baseComment dark"
      key="2"
    >
      <h5
        className="baseName dark"
      >
        mock_name
      </h5>
      <p
        className="dark"
      >
        mock_content
      </p>
      <p
        className="baseDate dark"
      >
        mock_date
      </p>
    </li>
  </ul>
</div>
`;

exports[`Comments should matchSnapshot when given an array of comments 1`] = `
<div>
  <ul
    className="baseCommentsList null"
  >
    <li
      className="baseComment null"
      key="0"
    >
      <h5
        className="baseName null"
      >
        mock_name
      </h5>
      <p
        className="null"
      >
        mock_content
      </p>
      <p
        className="baseDate null"
      >
        mock_date
      </p>
    </li>
    <li
      className="baseComment null"
      key="1"
    >
      <h5
        className="baseName null"
      >
        mock_name
      </h5>
      <p
        className="null"
      >
        mock_content
      </p>
      <p
        className="baseDate null"
      >
        mock_date
      </p>
    </li>
    <li
      className="baseComment null"
      key="2"
    >
      <h5
        className="baseName null"
      >
        mock_name
      </h5>
      <p
        className="null"
      >
        mock_content
      </p>
      <p
        className="baseDate null"
      >
        mock_date
      </p>
    </li>
  </ul>
</div>
`;

exports[`Comments should matchSnapshot when given an empty array of comments 1`] = `
<div>
  <ul
    className="baseCommentsList null"
  />
</div>
`;
